storage-access.bs

```diff
@@ -139,6 +139,15 @@ To <dfn type="abstract-op">generate a partitioned storage key</dfn> for a {{Docu
 1. Let |top-level site| be the result of [=obtain a site|obtaining a site=] from |settings|' [=top-level origin=].
 1. Return the [=partitioned storage key=] (|top-level site|, |site|).
 
+To <dfn type="abstract-op">generate an overridden partitioned storage key</dfn> for a {{Document}} |doc| and an [=url/origin=] |override|, run the following steps:
```

Comment:
In the latest commit, I tried `delegated`, though I am not sure it's better. 

Other options:
* split this into steps that are passed both URL arguments, i.e., the equivalent of `generate_partitioned_key(top_level_site, embedded_origin)`, and then have both methods use the new function, rather than taking a document. The downside would be that both `requestStorageAccess` and `requestStorageAccessForOrigin` would have steps to extract the URLs, whereas they can be combined somewhat this way.
* besides `delegated` and `overridden`, I suppose we could consider `forwarded` or something. My thesaurus lookups didn't help much, unfortunately.